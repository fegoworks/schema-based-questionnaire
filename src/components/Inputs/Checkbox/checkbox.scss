.Checkbox {
  margin-bottom: 0.6rem;
  display: flex;
  flex-direction: column;
}
.Checkbox label {
  cursor: pointer;
  display: inline-flex;
  align-items: center;
  font-weight: 500;

  span {
    margin-left: 0.5rem;
  }
}

.Checkbox input[type="checkbox"] {
  width: 2em;
  height: 2rem;
  accent-color: green;
}

.Checkbox .error {
  color: #d44;
  margin: 0;
}
