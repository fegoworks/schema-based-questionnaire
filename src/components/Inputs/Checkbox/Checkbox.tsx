import "./checkbox.scss";
import { FunctionComponent } from "react";
import { FieldProps } from "../../formField";

export type Opt = { value: string | number; label: string };

const Checkbox: FunctionComponent<FieldProps> = ({
  error,
  name,
  onBlur,
  onChange,
  touched,
  options,
  disabled,
}) => {
  return (
    <div className="Checkbox" role="checkbox">
      {options?.map((option) => (
        <label key={option.value}>
          <input
            onChange={onChange}
            onBlur={onBlur}
            type="checkbox"
            name={name}
            value={option.value}
            disabled={disabled}
          />
          <span>{option.label}</span>
        </label>
      ))}
      {error && touched && <p className="error">{error}</p>}
    </div>
  );
};

export default Checkbox;
