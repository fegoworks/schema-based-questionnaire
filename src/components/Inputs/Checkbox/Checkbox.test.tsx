import { fireEvent, render } from "@testing-library/react";
import { describe, expect, vi } from "vitest";
import Checkbox from "./Checkbox";

const opt = [
  { label: "Check", value: "check" },
  { label: "Text", value: "text" },
  { label: "Select", value: "select" },
];
const defaultProps = {
  name: "types",
  options: opt,
  disabled: false,
};

describe("Component - Checkbox", () => {
  let onChange: () => void;

  beforeEach(() => {
    onChange = vi.fn();
  });

  it("should render checkbox component", () => {
    const { container } = render(
      <Checkbox onChange={() => onChange} {...defaultProps} />
    );
    expect(container).toMatchSnapshot();
  });

  it("should render a checkbox for each option", () => {
    const { getByRole } = render(
      <Checkbox onChange={() => onChange} {...defaultProps} value="text" />
    );
    expect(getByRole("checkbox", { name: "Check" })).toBeInTheDocument();
    expect(getByRole("checkbox", { name: "Select" })).toBeInTheDocument();
    expect(getByRole("checkbox", { name: "Text" })).toBeInTheDocument();
  });

  it("should render a checkbox for each option", () => {
    const { getByRole } = render(
      <Checkbox onChange={() => onChange} {...defaultProps} />
    );
    const TextOption = getByRole("checkbox", { name: "Check" });

    expect(TextOption).not.toBeChecked();
    fireEvent.click(getByRole("checkbox", { name: "Check" }));
    expect(TextOption).toBeChecked();
  });

  it("should show an error message if validation is incorrect", () => {
    const { getByText } = render(
      <Checkbox
        onChange={() => onChange}
        {...defaultProps}
        error={"Some error"}
        touched={true}
      />
    );
    expect(getByText("Some error")).toBeTruthy();
  });
});
