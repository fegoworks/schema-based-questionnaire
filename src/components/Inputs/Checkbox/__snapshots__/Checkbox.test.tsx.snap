// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component - Checkbox > should render checkbox component 1`] = `
<div>
  <div
    class="Checkbox"
    role="checkbox"
  >
    <label>
      <input
        name="types"
        type="checkbox"
        value="check"
      />
      <span
        class=""
      >
        Check
      </span>
    </label>
    <label>
      <input
        name="types"
        type="checkbox"
        value="text"
      />
      <span
        class=""
      >
        Text
      </span>
    </label>
    <label>
      <input
        name="types"
        type="checkbox"
        value="select"
      />
      <span
        class=""
      >
        Select
      </span>
    </label>
  </div>
</div>
`;
